restli-tools/src/main/java/com/linkedin/restli/tools/data/ExtensionSchemaValidation.java

```diff
@@ -0,0 +1,150 @@
+
+  public static void main(String[] args) throws Exception
+  {
+    try {
+      final CommandLineParser parser = new GnuParser();
+      CommandLine cl = parser.parse(_options, args);
+
+      if (cl.hasOption('h'))
+      {
+        help();
+        System.exit(0);
+      }
+
+      String[] cliArgs = cl.getArgs();
+      if (cliArgs.length != 2)
+      {
+        _logger.error("Wrong argument given");
+        help();
+        System.exit(1);
+      }
+      int i = 0;
+      String resolverPath = cliArgs[i++];
+      String inputPath = cliArgs[i];
+
+      File inputDir = new File(inputPath);
+
+      if (!inputDir.exists() || !inputDir.canRead()) {
+        _logger.error("Input directory does not exist or cannot be read: " + inputDir.getAbsolutePath());
+        System.exit(1);
+      }
+
+      parseAndValidateExtensionSchemas(resolverPath, inputDir);
+    }
+    catch (ParseException e)
+    {
+      _logger.error("Invalid arguments: " + e.getMessage());
+      System.exit(1);
+    }
+  }
+
+  private static void parseAndValidateExtensionSchemas(String resolverPath, File inputDir) throws IOException
+  {
+    // Parse each extension schema and validate it
+    Iterator<File> iterator = FileUtils.iterateFiles(inputDir, new String[]{_pdl}, true);
+    while(iterator.hasNext())
+    {
+      File inputFile = iterator.next();
+      DataSchemaResolver resolver = MultiFormatDataSchemaResolver.withBuiltinFormats(resolverPath);
+      PdlSchemaParser parser = new PdlSchemaParser(resolver);
+      parser.parse(new FileInputStream(inputFile));
+      if (parser.hasError())
+      {
+        _logger.error(parser.errorMessage());
+        System.exit(1);
+      }
+
+      List<DataSchema> topLevelDataSchemas = parser.topLevelDataSchemas();
+      if (topLevelDataSchemas == null || topLevelDataSchemas.isEmpty() || topLevelDataSchemas.size() > 1)
+      {
+        _logger.error("Could not parse extension schema : " + inputFile.getAbsolutePath());
+        System.exit(1);
+      }
+      DataSchema topLevelDataSchema = topLevelDataSchemas.get(0);
+      if (!(topLevelDataSchema instanceof NamedDataSchema))
+      {
+        _logger.error("Invalid extension schema : " + inputFile.getAbsolutePath());
+        System.exit(1);
+      }
+      if (!((NamedDataSchema) topLevelDataSchema).getFullName().endsWith("Extensions"))
+      {
+        _logger.error("Invalid extension schema name : [{}]. The name of the extension schema must be <baseSchemaName> + 'Extensions'", ((NamedDataSchema) topLevelDataSchema).getFullName());
+        System.exit(1);
+      }
+
+      List<NamedDataSchema> includes = ((RecordDataSchema)topLevelDataSchema).getInclude();
+      // TODO: Check includes schemas can only be the resource schemas
+
+      List<RecordDataSchema.Field> extensionSchemaFields = ((RecordDataSchema)topLevelDataSchema).getFields().
+          stream().
+          filter(
+              f -> !((RecordDataSchema) topLevelDataSchema).isFieldFromIncludes(f))
```

Comment:
uncessary parenthesis, also can be add to last line, no need to split

Reply:
Can you please explain "unnecessary parenthesis"? I don't see any.